gcc/rust/checks/errors/rust-feature-gate.cc

```diff
@@ -39,25 +39,68 @@ FeatureGate::check (AST::Crate &crate)
 		option.parse_to_meta_item ());
 	      for (const auto &item : meta_item->get_items ())
 		{
-		  const auto &name = item->as_string ();
-		  auto tname = Feature::as_name (name);
+		  const auto &name_str = item->as_string ();
+		  auto tname = Feature::as_name (name_str);
 		  if (!tname.is_none ())
-		    valid_features.push_back (Feature::create (tname.get ()));
+		    {
+		      auto name = tname.get ();
+		      valid_features.insert (name);
+		    }
+
 		  else
 		    rust_error_at (item->get_locus (), "unknown feature '%s'",
-				   name.c_str ());
+				   name_str.c_str ());
 		}
 	    }
 	}
     }
-  valid_features.shrink_to_fit ();
 
-  // TODO (mxlol233): add the real feature gate stuff.
   auto &items = crate.items;
   for (auto it = items.begin (); it != items.end (); it++)
     {
       auto &item = *it;
       item->accept_vis (*this);
     }
 }
+
+void
+FeatureGate::gate (Feature::Name name, Location loc,
+		   const std::string error_msg)
+{
+  if (!valid_features.count (name))
+    {
+      auto feature = Feature::create (name);
+      auto issue = feature.issue ();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-diag"
```

Comment:
I don't think this is necessary and we shouldn't lose the format string checker here. Is the issue with the URL you put in the format string?

Reply:
Also, this might cause issues when compiling `gccrs` with other compilers.